src/index.js

```diff
@@ -24,15 +24,28 @@ const containsNodeModules = (resolvedResource) => {
     return resolvedResource.includes('node_modules');
 };
 
-const findDuplicate = (res) => (t) => {
-    return res.includes(t);
-};
-
-const findBestMatch = (arr, matcher) => {
-    return arr.filter(matcher).sort((a, b) => b.length - a.length)[0];
+const findBestMatch = (key, prefixGroup, previousLock, currentLock, resolvedResource) => {
+    for (const prefix of prefixGroup) {
+        if (resolvedResource.includes(prefix)) {
+            // If we have a lock file. Always use the entry saved previously to achieve the long term caching.
+            if (key in previousLock) {
+                currentLock[key] = prefixGroup.find((prefix) => prefix.includes(previousLock[key]));
+            }
+
+            // Don't replace on the first encounter but assign the found prefix to the group.
+            // Next time if we found a prefix match we use the assigned result from previous iterations.
+            if (key in currentLock) {
+                return [prefix, currentLock[key]];
+            } else {
+                currentLock[key] = prefix;
+                return null;
+            }
+        }
+    }
+    return null;
 };
 
-const deduplicate = (result, dupVals) => {
+const deduplicate = (result, prefixGroups, previousLock, currentLock) => {
```

Comment:
`prefixGroups` is a confusing name to me - it's the `Object.entries` from `duplicates` isn't it? Why not just stick with the duplicates name for the param?

Reply:
Well. They are prefixes of duplicates. I can change it back.